worker/uniter/runner/context/context.go

```diff
@@ -352,6 +352,12 @@ type HookContext struct {
 	// secretChanges records changes to secrets during a hook execution.
 	secretChanges *secretsChangeRecorder
 
+	// noticeType is the Pebble notice type associated with the hook.
+	noticeType string
+
+	// noticeKey is the Pebble notice key associated with the hook.
+	noticeKey string
```

Comment:
Could we group the pebble attributes together? We have workloadName above.

Reply:
Done.